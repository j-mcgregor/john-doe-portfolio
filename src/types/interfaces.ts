export interface NavbarProps {
    brand: string | JSX.Element
}

export interface PRISMIC_RichTextType {
    type: string
    text: string
    spans: any[]
}

export interface PRISMIC_Image {
    dimensions: {
        width: number
        height: number
    }
    alt?: string | null
    copyright: string | null
    url: string
}

export interface PRISMIC_ExperienceType {
    job_title: string
    date_from: string
    date_to?: string
    present?: boolean
    company: string
    city: string
    description: PRISMIC_RichTextType[]
}

export interface PRISMIC_landing_node {
    node: {
        primary_text: PRISMIC_RichTextType[]
        secondary_text: PRISMIC_RichTextType[]
        background_image: PRISMIC_Image
        about: PRISMIC_RichTextType[]
        experience: PRISMIC_ExperienceType[]
    }
}

export interface IndexPageProps {
    data: {
        prismic: {
            allLandings: {
                edges: PRISMIC_landing_node[]
            }
        }
        file: {
            childImageSharp: {
                fixed: {
                    height: number
                    width: number
                    base64: string
                    src: string
                    srcSet: string
                }
            }
        }
    }
}

export interface HeroProps {
    primary: JSX.Element
    secondary: JSX.Element
    logo?: JSX.Element
    backgroundImage?: string
}

export interface StyledHeroProps {
    bgImageUrl?: string
}

export interface AboutProps {
    about: JSX.Element
    experience: JSX.Element[]
}

// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>/  FORMS   >>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

export interface TextInputProps {
    type: string
    placeholder: string
    label?: string
    name: string
    required: boolean
    classNames?: string[]
    value: string
    onChange: (e: React.FormEvent<HTMLInputElement>) => void
}

export interface TextAreaInputProps {
    placeholder: string
    label?: string
    name: string
    required: boolean
    value: string
    onChange: (e: React.FormEvent<HTMLTextAreaElement>) => void
}
