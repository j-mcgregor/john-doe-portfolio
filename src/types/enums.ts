export enum Breakpoints {
    xl = 1200,
    lg = 992,
    md = 768,
    sm = 576,
}

export enum GridStyle {
    GRID = 'GRID',
    ROW = 'ROW',
    COL = 'COL',
}
