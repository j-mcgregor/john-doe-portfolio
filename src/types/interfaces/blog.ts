// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>  BLOG   >>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

import {
    PRISMIC_RichTextType,
    PRISMIC_Image,
    PRISMIC_blog_node,
    PRISMIC_blog_node_body,
} from './prismic'

export interface PRISMIC_PRIMARY_Image_with_Caption {
    type: string
    label?: string
    primary?: {
        caption?: PRISMIC_RichTextType[]
        image?: PRISMIC_Image
    }
}

export interface PRISMIC_PRIMARY_Text {
    type: string
    label?: string
    primary?: {
        text?: PRISMIC_RichTextType[]
    }
}

export interface PRISMIC_PRIMARY_Quote {
    type: string
    label?: string
    primary?: {
        quote?: PRISMIC_RichTextType[]
    }
}

export interface BlogPageProps {
    data: {
        prismic: {
            allBlog_posts: {
                edges: PRISMIC_blog_node[]
            }
        }
    }
}

export interface BlogBodyProps {
    body: Array<
        | PRISMIC_PRIMARY_Image_with_Caption
        | PRISMIC_PRIMARY_Quote
        | PRISMIC_PRIMARY_Text
    >
}

export interface PostSummaryProps {
    post: PRISMIC_blog_node_body
}

export interface BlogPostsProps {
    posts?: PRISMIC_blog_node[]
}

export interface ImageCaptionProps {
    slice: PRISMIC_PRIMARY_Image_with_Caption
}
