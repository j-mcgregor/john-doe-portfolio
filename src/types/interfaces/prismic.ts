// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>/  PRISMIC   >>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

import { BlogBodyProps } from './blog'

export interface PRISMIC_RichTextType {
    spans: any[]
    text: string
    type: string
}

export interface PRISMIC_Image {
    alt?: string | null
    copyright: string | null
    dimensions: {
        height: number
        width: number
    }
    url: string
    thumbnail?: {
        dimensions: {
            width: number
            height: number
        }
        alt: string | null
        copyright: string | null
        url: string
    }
}

export interface PRISMIC_ExperienceType {
    city: string
    company: string
    date_from: string
    date_to?: string
    description: PRISMIC_RichTextType[]
    job_title: string
    present?: boolean
}

export interface PRISMIC_SocialLinks {
    name: string
    text?: string | null
    url?: {
        url: string
    } | null
}

export interface PRISMIC_gallery_body {
    type: string
    label?: string
    fields: PRISMIC_gallery_image_fields[]
}

export interface PRISMIC_gallery_image_fields {
    alt_text: PRISMIC_RichTextType[] | null
    image: PRISMIC_Image
    caption: PRISMIC_RichTextType[] | null
}

// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>/  NODES   >>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

export interface PRISMIC_landing_node {
    node: {
        about: PRISMIC_RichTextType[]
        background_image: PRISMIC_Image
        experience: PRISMIC_ExperienceType[]
        primary_text: PRISMIC_RichTextType[]
        secondary_text: PRISMIC_RichTextType[]
    }
}

export interface PRISMIC_contact_node {
    node: {
        title: PRISMIC_RichTextType[] | null
        subtitle: PRISMIC_RichTextType[] | null
        social_links: PRISMIC_SocialLinks[]
    }
}

export interface PRISMIC_gallery_node {
    node: {
        title: PRISMIC_RichTextType[] | null
        description: PRISMIC_RichTextType[] | null
        body: PRISMIC_gallery_body[]
    }
}

export interface PRISMIC_blog_node_body {
    title: PRISMIC_RichTextType[]
    date: string
    body: BlogBodyProps[]
    _meta: {
        id: string
        uid: string
        type: string
    }
}
export interface PRISMIC_blog_node {
    node: PRISMIC_blog_node_body
}
