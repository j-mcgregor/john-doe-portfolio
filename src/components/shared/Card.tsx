import React from 'react'
import RichText from '../../utils/RichTextCustom'
import { PRISMIC_ExperienceType } from '../../types/interfaces/prismic'

export interface CardProps {
    item: PRISMIC_ExperienceType
}

const Card: React.FC<CardProps> = ({ item }) => {
    return (
        <div className="mb3">
            <div className="text-center">
                <h5 className="m0">
                    {item.job_title} @ {item.company}
                </h5>
                <div className="my1">{item.city}</div>
                <small>
                    {item.date_from} to{' '}
                    {item.present ? 'present' : item.date_to}
                </small>
            </div>
            <RichText render={item.description} />
        </div>
    )
}

export default Card
