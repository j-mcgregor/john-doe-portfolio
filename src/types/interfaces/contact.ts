// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>/  CONTACT   >>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

import { PRISMIC_contact_node } from './prismic'

export interface ContactPageProps {
    data: {
        prismic: {
            allContacts: {
                edges: PRISMIC_contact_node[]
            }
        }
    }
}
