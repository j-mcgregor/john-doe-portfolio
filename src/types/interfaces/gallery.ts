// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>/  GALLERY >>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

import { PRISMIC_gallery_node, PRISMIC_gallery_image_fields } from './prismic'

export interface GalleryContainerProps {
    images: PRISMIC_gallery_image_fields[]
}

export interface LocalFileNode {
    node: {
        name: string
        id: string
        publicURL: string
    }
}

export interface GalleryPageProps {
    data: {
        prismic: {
            allGallerys: {
                edges: PRISMIC_gallery_node[]
            }
        }
        allFile: {
            edges: LocalFileNode[]
        }
    }
}

export interface SVG_Node {
    node: {
        extension: string
        id: string
        internal: {
            mediaType: string
        }
        name: string
        publicURL: string
    }
}

export interface SVG_StaticRequestProps {
    allFile: {
        edges: SVG_Node[]
    }
}
