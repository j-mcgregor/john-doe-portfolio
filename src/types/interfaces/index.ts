// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>/  INDEX   >>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

import { PRISMIC_landing_node } from './prismic'

export interface IndexPageProps {
    data: {
        prismic: {
            allLandings: {
                edges: PRISMIC_landing_node[]
            }
        }
        file: {
            childImageSharp: {
                fixed: {
                    height: number
                    width: number
                    base64: string
                    src: string
                    srcSet: string
                }
            }
        }
    }
}
