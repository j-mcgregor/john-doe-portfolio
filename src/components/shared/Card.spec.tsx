import React from 'react'
import { render } from '@testing-library/react'
import Card from './Card'
import { PRISMIC_ExperienceType } from '../../types/interfaces'

describe('Card', () => {
    let item: PRISMIC_ExperienceType

    beforeEach(() => {
        item = {
            job_title: 'Dog walker',
            date_from: '2000-10-10',
            date_to: '2010-10-10',
            company: 'Walkies!?',
            city: 'The Park',
            description: [
                {
                    text: 'Goofy Goober',
                    type: 'header3',
                    spans: [],
                },
            ],
        }
    })
    it('should render', () => {
        const component = render(<Card item={item} />)

        expect(component).toMatchSnapshot()
    })

    it('should display "present" if present field is true', () => {
        const itemWithPresentField: PRISMIC_ExperienceType = {
            ...item,
            present: true,
        }
        const component = render(<Card item={itemWithPresentField} />)

        expect(component.findByText(/present/)).toBeDefined()
        expect(component).toMatchSnapshot()
    })
})
