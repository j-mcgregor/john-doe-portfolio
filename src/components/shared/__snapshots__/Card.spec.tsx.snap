// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should display "present" if present field is true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mb3"
      >
        <div
          class="text-center"
        >
          <h5
            class="m0"
          >
            Dog walker
             @ 
            Walkies!?
          </h5>
          <div
            class="my1"
          >
            The Park
          </div>
          <small>
            2000-10-10
             to
             
            present
          </small>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mb3"
    >
      <div
        class="text-center"
      >
        <h5
          class="m0"
        >
          Dog walker
           @ 
          Walkies!?
        </h5>
        <div
          class="my1"
        >
          The Park
        </div>
        <small>
          2000-10-10
           to
           
          present
        </small>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mb3"
      >
        <div
          class="text-center"
        >
          <h5
            class="m0"
          >
            Dog walker
             @ 
            Walkies!?
          </h5>
          <div
            class="my1"
          >
            The Park
          </div>
          <small>
            2000-10-10
             to
             
            2010-10-10
          </small>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mb3"
    >
      <div
        class="text-center"
      >
        <h5
          class="m0"
        >
          Dog walker
           @ 
          Walkies!?
        </h5>
        <div
          class="my1"
        >
          The Park
        </div>
        <small>
          2000-10-10
           to
           
          2010-10-10
        </small>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
